File: fix_paths.py
import os

def fix_paths():
    base_dir = r"d:\WORK\3_tasks\1_TASK\tanusha\pages"
    count = 0
    # Обходим все файлы в папке pages
    for root, _, files in os.walk(base_dir):
        for file in files:
            if file.endswith('.php'):
                path = os.path.join(root, file)
                with open(path, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                # Убираем лишние require_once
                new_content = content.replace("require_once 'config.php';", "")
                new_content = new_content.replace("require_once 'auth.php';", "")
                
                if content != new_content:
                    with open(path, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    print(f"Исправлен: {file}")
                    count += 1

    # Исправляем header.php
    header_path = r"d:\WORK\3_tasks\1_TASK\tanusha\header.php"
    if os.path.exists(header_path):
        with open(header_path, 'r', encoding='utf-8') as f:
            content = f.read()
        new_content = content.replace("require_once 'auth.php';", "")
        if content != new_content:
            with open(header_path, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print("Исправлен: header.php")
            count += 1
            
    print(f"Готово! Исправлено файлов: {count}")

if __name__ == "__main__":
    fix_paths()

File: fix_paths.php
<?php
// fix_paths.php - Загрузите на сервер и откройте в браузере
$dir = __DIR__ . '/pages';
$iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($dir));

echo "<h3>Исправление путей...</h3>";
$count = 0;
foreach ($iterator as $file) {
    if ($file->isFile() && $file->getExtension() === 'php') {
        $content = file_get_contents($file->getPathname());
        $new_content = str_replace("require_once 'config.php';", "", $content);
        $new_content = str_replace("require_once 'auth.php';", "", $new_content);
        if ($content !== $new_content) {
            file_put_contents($file->getPathname(), preg_replace("/^\\s*$/m", "", $new_content));
            echo "Исправлен: " . $file->getFilename() . "<br>";
            $count++;
        }
    }
}

// Исправление header.php
$headerFile = __DIR__ . '/header.php';
$content = file_get_contents($headerFile);
$new_content = str_replace("require_once 'auth.php';", "", $content);
if ($content !== $new_content) {
    file_put_contents($headerFile, $new_content);
    echo "Исправлен: header.php<br>";
    $count++;
}

echo "<br><b>Готово! Исправлено файлов: $count</b>";
?>
